hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/CapacityScheduler.java

```diff
@@ -3366,14 +3367,47 @@ public boolean moveReservedContainer(RMContainer toBeMovedContainer,
 
   @Override
   public long checkAndGetApplicationLifetime(String queueName,
```

Comment:
I think instead of add this create new queue logic to the method we could make the getOrCreateQueueFromPlacementContext public and call it before we call the checkAndGetApplicationLifetime method. If i see well this checkAndGetApplicationLifetime used just once in the code